tt_metal/llrt/tt_cluster.cpp

```diff
@@ -49,27 +49,26 @@ Cluster::Cluster() {
 }
 
 void Cluster::detect_arch_and_target() {
-#ifdef TT_METAL_VERSIM_DISABLED
-    this->target_type_ = TargetDevice::Silicon;
-    std::vector<chip_id_t> physical_mmio_device_ids = tt_SiliconDevice::detect_available_device_ids();
-    this->arch_ = detect_arch(physical_mmio_device_ids.at(0));
-    for (int dev_index = 1; dev_index < physical_mmio_device_ids.size(); dev_index++) {
-        chip_id_t device_id = physical_mmio_device_ids.at(dev_index);
-        tt::ARCH detected_arch = detect_arch(device_id);
-        TT_FATAL(
-            this->arch_ == detected_arch,
-            "Expected all devices to be {} but device {} is {}",
-            get_arch_str(this->arch_),
-            device_id,
-            get_arch_str(detected_arch));
+    if(std::getenv("TT_METAL_SIMULATOR_EN")) {
+        this->target_type_ = TargetDevice::Simulator;
+        auto arch_env = getenv("ARCH_NAME");
```

Comment:
if we do ARCH_NAME for simulator wouldn't we need SIMULATOR_GS/SIMULATOR_WH/SIMULATOR_BH, ... etc?

Reply:
Yes with `ARCH_NAME` we shoulnd't need anything like `SIMULATOR_GS`. 

Technically right now we don't need ARCH_NAME at all since we only support blackhole, and in the future we would only be using the simulator for quasar, so this could be hardcoded?

Reply:
where is ARCH_NAME consumed in the simulator flow?

Reply:
Mainly consumed by `get_soc_description_file` and `get_core_descriptor_file` to get the corresponding descriptor yaml.